pkg/graphql/graphql.go

```diff
@@ -944,3 +956,84 @@ func (r *Resolver) Logs(ctx context.Context, args struct{ Filter FilterCriteria
 	filter := filters.NewRangeFilter(filters.Backend(r.backend), begin, end, addresses, topics)
 	return runFilter(ctx, r.backend, filter)
 }
+
+// StorageResult represents a storage slot value. All arguments are mandatory.
+type StorageResult struct {
+	value     []byte
+	cid       string
+	ipldBlock []byte
+}
+
+func (s *StorageResult) Value(ctx context.Context) common.Hash {
+	return common.BytesToHash(s.value)
+}
+
+func (s *StorageResult) Cid(ctx context.Context) string {
+	return s.cid
+}
+
+func (s *StorageResult) IpldBlock(ctx context.Context) hexutil.Bytes {
+	return hexutil.Bytes(s.ipldBlock)
+}
+
+func (r *Resolver) GetStorageAt(ctx context.Context, args struct {
+	BlockHash common.Hash
+	Contract  common.Address
+	Slot      common.Hash
+}) (*StorageResult, error) {
+	cid, ipldBlock, rlpValue, err := r.backend.IPLDRetriever.RetrieveStorageAtByAddressAndStorageSlotAndBlockHash(args.Contract, args.Slot, args.BlockHash)
+
+	if err != nil {
+		if err == sql.ErrNoRows {
+			ret := StorageResult{value: []byte{}, cid: "", ipldBlock: []byte{}}
+
+			return &ret, nil
+		}
+
+		return nil, err
+	}
+
+	var value interface{}
+	err = rlp.DecodeBytes(rlpValue, &value)
+	if err != nil {
+		return nil, err
+	}
+
+	ret := StorageResult{value: value.([]byte), cid: cid, ipldBlock: ipldBlock}
+	return &ret, nil
+}
+
+func (r *Resolver) GetLogs(ctx context.Context, args struct {
+	BlockHash common.Hash
+	Contract  common.Address
+}) (*[]*Log, error) {
+	ret := make([]*Log, 0, 10)
+
+	receiptCIDs, receiptsBytes, err := r.backend.IPLDRetriever.RetrieveReceiptsByBlockHash(args.BlockHash)
+	if err != nil {
+		return nil, err
+	}
+
+	receipts := make(types.Receipts, len(receiptsBytes))
+	for index, receiptBytes := range receiptsBytes {
+		receiptCID := receiptCIDs[index]
+		receipt := new(types.Receipt)
+		if err := rlp.DecodeBytes(receiptBytes, receipt); err != nil {
```

Comment:
For compatibility with EIP-2718 enveloped transactions and receipts (EIP-2930 access list and EIP-1559 dynamic fee) we can no longer use RLP encoding/decoding, we instead need to use the `MarshalBinary` and `UnmarshalBinary` methods.

Related issue and more info here: https://github.com/vulcanize/ipld-eth-server/issues/78#issuecomment-873616604.